Lib/gftools/builder/_ninja.py

```diff
@@ -61,6 +62,23 @@ def build(self):
             if os.path.exists(temporary):
                 os.remove(temporary)
 
+        # Clean up temp build files
+        search_directory = os.getcwd()
+        target_names = ["build.ninja", ".ninja_log", "instance_ufo", "master_ufo"]
+        for root, dirs, files in os.walk(search_directory, topdown=False):
+            for file in files:
+                if file in target_names:
+                    file_path = os.path.join(root, file)
+                    os.remove(file_path)
+                    print(f"Removed file: {file_path}")
+        for dir_name in dirs:
```

Comment:
Shouldn't this be within the "os.walk" loop?

Reply:
Sure, like this? I tested this with a few different font repos and it seems to work, and pushed it as an update to this PR (e558616).
```python
        # Clean up temp build files
        search_directory = os.getcwd()
        target_names = ["build.ninja", ".ninja_log", "instance_ufo", "master_ufo"]
        for root, dirs, files in os.walk(search_directory, topdown=False):
            for file in files:
                if file in target_names:
                    file_path = os.path.join(root, file)
                    os.remove(file_path)
                    print(f"Removed file: {file_path}")
            for dir_name in dirs[:]:
                if dir_name in target_names:
                    dir_path = os.path.join(root, dir_name)
                    shutil.rmtree(dir_path)
                    print(f"Removed directory: {dir_path}")
```

Reply:
The shallow copy of dirs `dirs[:]` was ChatGPTs idea, but I looked into it and it made sense to me. ChatGPTs explanation:
>By creating a copy with dirs[:], you can safely remove elements from the original dirs list within the loop without affecting the ongoing iteration. This is particularly useful when you want to conditionally remove items from a list while iterating over it.

Reply:
Nice, thanks.